rpc/backend.go

```diff
@@ -0,0 +1,179 @@
+package rpc
+
+import (
+	"fmt"
+	"math/big"
+	"strconv"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/ethermint/x/evm"
+	"github.com/cosmos/ethermint/x/evm/types"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+// Backend implements the functionality needed to filter changes.
+// Implemented by PublicEthAPI.
+type Backend interface {
```

Comment:
does geth has this interface too? Maybe we can use that instead?

Reply:
If not, can you add a comment on top of the interface def?

Reply:
geth's backend is similar but it has subscriptions using channels, we could implement something similar perhaps, but we would have to define our own "events" instead
https://github.com/ethereum/go-ethereum/blob/d90d1db609c8d77baa422d49bd371207c06b4711/eth/filters/filter.go#L33

Reply:
Just to note it may make it easier in the future to use/ implement the Ethereum interface if you want to take advantage of some of their utility functions which have this backend as a parameter (iirc, just going off memory)